odl/contrib/shearlab/shearlab_operator.py

```diff
@@ -0,0 +1,324 @@
+# Copyright 2014-2018 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""ODL integration with shearlab."""
+
+import odl
+import numpy as np
+from threading import Lock
+
+# Library for the shearlab library
+
+import julia
+import matplotlib.pyplot as plt
+from numpy import ceil
+from numpy.fft import fft2, ifft2, fftshift, ifftshift
+
+__all__ = ('ShearlabOperator',)
+
+
+class ShearlabOperator(odl.Operator):
+
+    """Shearlet transform using Shearlab.jl as backend.
+    This is the non-compact shearlet transform implemented using the Fourier
+    transform.
+    """
+
+    def __init__(self, space, num_scales):
+
+        """Initialize a new instance.
+        Parameters
+        ----------
+        space : `DiscreteLp`
+            The space on which the shearlet transform should act. Must be
+            two-dimensional.
+        num_scales : nonnegative `int`
+            The number of scales for the shearlet transform, higher numbers
+            mean better edge resolution but more computational burden.
+        Examples
+        --------
```

Comment:
Same here: each of these section headings need a blank line above them, otherwise they're not recognized.